### PR TITLE
Angular2

### PR DESCRIPTION
**Pull request by ionkoto**
Angular2 frontend added to the project. 
Added Event logging functionality to the backend that logs events on DB interactions.
The events get pooled and handled in the frontend to respond on DB changes, making the app more reactive.
The com.fasterxml.jackson.core dependency added to the POM is used for parsing Enum types to JSON.

### CODE REVIEW

File: com.musala.simple.students.spring/src/main/angular/src/app/add-student/add-student.component.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        }, 
+        err => {
+          console.log(err);
+        });
+  }
+  constructor(studentService: StudentsService, private router: Router) {
+    this.studentService = studentService;
+  }
+
+  ngOnInit() {
```
Comment (alex-kostadinov) [inline com.musala.simple.students.spring/src/main/angular/src/app/add-student/add-student.component.ts]: Do we need this empty method here? :)

File: com.musala.simple.students.spring/src/main/angular/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import {NgModule} from "@angular/core";
+import { RouterModule } from "@angular/router";
+
+import {ListComponent} from "./list/list.component";
+import { StudentComponent } from "./student/student.component";
+import { AddStudentComponent } from "./add-student/add-student.component";
+
+import { AddStudentModule } from './add-student/add-student.module';
+
+const routes = [
```
Comment (alex-kostadinov) [inline com.musala.simple.students.spring/src/main/angular/src/app/app-routing.module.ts:10]: Beautify the formatting of these routes :)

File: com.musala.simple.students.spring/src/main/angular/src/app/header/header.component.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.css']
+})
+export class HeaderComponent implements OnInit {
+
+  constructor() { }
```
Comment (alex-kostadinov) [inline com.musala.simple.students.spring/src/main/angular/src/app/header/header.component.ts]: Are these empty constructor and method needed?

File: com.musala.simple.students.spring/src/main/angular/src/app/item/item.component.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  studentsService: StudentsService;
+  router: Router;
+  subscription: Subscription;
+
+  constructor(studentsService: StudentsService, router: Router, private modalService: BsModalService) {
+    this.studentsService = studentsService;
+    this.router = router;
+  }
+
+  ngOnInit() {
```
Comment (alex-kostadinov) [inline com.musala.simple.students.spring/src/main/angular/src/app/item/item.component.ts]: Check this too.

File: com.musala.simple.students.spring/src/main/angular/src/app/list/list.component.html [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+      <label class="btn btn-primary" [class.active]="myForm.value.mysql" btnCheckbox formControlName="mysql">MySql</label>
+    </div>
+  </form>
+</div>
+<!-- 
+
+<div class="container">
+  <div class="row">
+    <div class="col-xs-6">
```
Comment (alex-kostadinov) [inline com.musala.simple.students.spring/src/main/angular/src/app/list/list.component.html]: Please removed commented-out markup and check formatting.

File: com.musala.simple.students.spring/src/main/angular/src/app/students.service.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+          this.mongoStudentsChanged.next();
+          break;
+        }
+        case "mysql": {
+          this.studentsMysql.unshift(newStudent);
+          this.mysqlStudentsChanged.next();
+          break;
+        }
+      }
+      //this.studentsChanged.next();
```
Comment (alex-kostadinov) [inline com.musala.simple.students.spring/src/main/angular/src/app/students.service.ts]: Delete this.

File: com.musala.simple.students.spring/src/main/angular/src/index.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<!doctype html>
```
Comment (alex-kostadinov) [inline com.musala.simple.students.spring/src/main/angular/src/index.html]: Format this markup.

File: com.musala.simple.students.spring/src/main/java/com/musala/simple/students/spring/web/EventsController.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.musala.simple.students.spring.web.dbevents.Event;
+
+@RestController
+public class EventsController {
+
+    @GetMapping(value = "/events")
+    @CrossOrigin(origins = "http://localhost:4200")
```
Comment (alex-kostadinov) [inline com.musala.simple.students.spring/src/main/java/com/musala/simple/students/spring/web/EventsController.java]: Can you extract this to a constant or, if it is to be changed, a configuration.properties file? :)

**Review by alex-kostadinov — CHANGES_REQUESTED** [review]

File: com.musala.simple.students.spring/src/main/angular/src/app/add-student/add-student.component.ts (hunk 1 above)
Comment (ionkoto) [reply com.musala.simple.students.spring/src/main/angular/src/app/add-student/add-student.component.ts]: no :)

**Review by alex-kostadinov — APPROVED** [review]

**Pull request merged**